packages/inngest/src/components/InngestCommHandler.ts

```diff
@@ -1451,6 +1441,134 @@ export class InngestCommHandler<
     return body;
   }
 
+  protected async inBandRegisterBody({
+    actions,
+    deployId,
+    signatureValidation,
+    url,
+  }: {
+    actions: HandlerResponseWithErrors;
+
+    /**
+     * Non-optional to ensure we always consider if we have a deploy ID
+     * available to us to use.
+     */
+    deployId: string | undefined | null;
+
+    signatureValidation: ReturnType<InngestCommHandler["validateSignature"]>;
+
+    url: URL;
+  }): Promise<InBandRegisterRequest> {
+    const registerBody = this.registerBody({ deployId, url });
+    const introspectionBody = await this.introspectionBody({
+      actions,
+      signatureValidation,
+      url,
+    });
+
+    if (!introspectionBody.authentication_succeeded) {
+      throw new Error("Cannot introspect without successful authentication");
+    }
+
+    const body: InBandRegisterRequest = {
+      app_id: this.client.id,
+      capabilities: registerBody.capabilities,
+      env: introspectionBody.env,
+      framework: registerBody.framework,
+      functions: registerBody.functions,
+      inspection: introspectionBody,
+      platform: "?",
```

Comment:
Is it possible to populate this?

Reply:
Baha yes, this is prior to our discussion. Will change.